xds/src/test/java/io/grpc/xds/FakeControlPlaneXdsIntegrationTest.java

```diff
@@ -214,15 +82,7 @@ serverHostName, clientListener(serverHostName)
 
   @Test
   public void pingPong_metadataLoadBalancer() throws Exception {
-    String tcpListenerName = SERVER_LISTENER_TEMPLATE_NO_REPLACEMENT;
-    String serverHostName = "test-server";
-    controlPlaneService.setXdsConfig(ADS_TYPE_URL_LDS, ImmutableMap.of(
-        tcpListenerName, serverListener(tcpListenerName),
-        serverHostName, clientListener(serverHostName)
-    ));
-    startServer(defaultBootstrapOverride());
-    controlPlaneService.setXdsConfig(ADS_TYPE_URL_RDS,
-        ImmutableMap.of(RDS_NAME, rds(serverHostName)));
+    LoadBalancerRegistry.getDefaultRegistry().register(new MetadataLoadBalancerProvider());
```

Comment:
Why not deregister of the LoadBalancerProvider at the end of the test?

Reply:
I was just trying to avoid the extra try/finally blocks since deregistering is not technically necessary. But I guess it is neater, I added it now.